test/lambda-handler-tests.js

```diff
@@ -110,410 +108,126 @@ describe('lambda-handler-as-promised', function() {
-							expect(err).to.deep.equal(testError);
-							done();
-						} catch (assertErr) {
-							done(assertErr);
-						}
-					});
-				});
+			const fixture = lambdaHandler().use(spy).use(mock);
 
-				it('should use value returned by "onError" as a result', function(done) {
-					const testResult = { testResponse: 'some useful info' };
-					const fixture = lambdaHandler(
-						() => { throw new Error(); },
-						{ onError: () => testResult }
-					);
-
-					fixture({}, testContext, (err, result) => {
-						try {
-							expect(err).to.be.equal(null);
-							expect(result).to.deep.equal(testResult);
-							done();
-						} catch (assertErr) {
-							done(assertErr);
-						}
-					});
+			return fixture(testEvent, testContext, noop)
+				.then(() => {
+					expect(spy.calledOnce).to.be.true;
+					mock.verify();
 				});
-
-				it('should use value returned by "onError" as a result when value is falsy', function(done) {
-					const testResult = false;
-					const fixture = lambdaHandler(
-						() => { throw new Error(); },
-						{ onError: () => testResult }
-					);
-
-					fixture({}, testContext, (err, result) => {
-						try {
-							expect(err).to.be.equal(null);
-							expect(result).to.deep.equal(testResult);
-							done();
-						} catch (assertErr) {
-							done(assertErr);
-						}
-					});
-				});
-
-				it('should return original error when "onError" does not return any value and does not throw', function(done) {
-					const testError = new Error('Winter is coming!');
-					const fixture = lambdaHandler(
-						() => { throw testError; },
-						{ onError: () => {} }
-					);
-
-					fixture({}, testContext, (err, result) => {
-						try {
-							expect(err).to.be.equal(null);
-							expect(result).to.deep.equal(testError);
-							done();
-						} catch (assertErr) {
-							done(assertErr);
-						}
-					});
-				});
-
-				it('should call "onAfter" hook when result is returned in "onError"', function(done) {
-					let resultHookCallCount = 0;
-					const testResult = { testResponse: 'some useful info' };
-					const fixture = lambdaHandler(
-						() => { throw new Error(); },
-						{
-							onError: () => testResult,
-							onAfter: result => {
-								expect(result).to.be.deep.equal(testResult);
-								resultHookCallCount++;
-							}
-						}
-					);
-
-					fixture({}, testContext, (err, result) => {
-						try {
-							expect(err).to.be.equal(null);
-							expect(resultHookCallCount).to.equal(1);
-							expect(result).to.be.deep.equal(testResult);
-							done();
-						} catch (assertErr) {
-							done(assertErr);
-						}
-					});
-				});
-			});
 		});
 
-		describe('errorStack', function() {
-			it('should throw when "errorStack" is not boolean', function() {
-				expect(() => lambdaHandler(() => {}, { errorStack: 'yes' }))
-					.to.throw(Error, /^options.errorStack must be a boolean when present$/);
+		it('lets middleware catch errors', function() {
+			const mock = sinon.mock();
```

Comment:
Feels messy to have spy, stub and mock in one test. Especially when mock does not check anything. Mocks are useful sometimes, but my experience with them is not that great. According to the docs, mocks should only be used when you need both: validate input and stub behaviour. Otherwise, just use spies or stubs.